applications/nrf5340_audio/src/bluetooth/bt_management/bt_mgmt.c

```diff
@@ -199,8 +204,13 @@ static void disconnected_cb(struct bt_conn *conn, uint8_t reason)
 
 	if (IS_ENABLED(CONFIG_BT_CENTRAL)) {
 		ret = bt_mgmt_scan_start(0, 0, BT_MGMT_SCAN_TYPE_CONN, NULL, BRDCAST_ID_NOT_USED);
-		if (ret) {
-			LOG_ERR("Failed to restart scanning: %d", ret);
+		if (ret && ret != -EALREADY) {
+			if (ret == -EIO) {
+				LOG_WRN("Controller might already in scanning state, check the HCI "
```

Comment:
```suggestion
				LOG_WRN("Controller might already be scanning, check the HCI "
```

Reply:
What is meant by check HCI?

Reply:
Not sure we should mention HCI at all. This is a common debugging step.

Reply:
There's a chance the log will print like this:
<wrn> bt_hci_core: opcode 0x2041 status 0x0c
0x2041 means scanning, 0x0c means already.
The host doesn't print low level response from controller, instead, the host only return -EIO in this case.

Reply:
#define BT_HCI_ERR_CMD_DISALLOWED   0x0c
But we need to figure out why we are trying to get the scanner to scan while it's already scanning

Reply:
Just figured out the root cause:
If two headsets are reset at a very close time, the disconnect_cb() from gateway will be triggered twice all most at the ver y close time. So bt_mgmt_scan_start() might be executed twice closely, So the check mechanism might failed leads to duplicate scanning.